modules/gapi/src/streaming/onevpl/engine/transcode/transcode_engine_legacy.cpp

```diff
@@ -0,0 +1,477 @@
+                                  CfgParam::vpp_in_crop_x_name() <<
+                                  "\": " << mfxVPPParams.vpp.In.CropX <<
+                                  " must be less or equal to \"" <<
+                                  CfgParam::vpp_in_width_name() << "\": " <<
+                                  mfxVPPParams.vpp.In.Width);
+        GAPI_Assert(false && "Invalid VPP params combination: Width & Crop");
+    }
+
+    if (mfxVPPParams.vpp.In.Height < mfxVPPParams.vpp.In.CropH + mfxVPPParams.vpp.In.CropY) {
+        GAPI_LOG_WARNING(nullptr, "Invalid vonfiguration params: sum \"" <<
+                                  CfgParam::vpp_in_crop_h_name() <<
+                                  "\": " << mfxVPPParams.vpp.In.CropH << " and \"" <<
+                                  CfgParam::vpp_in_crop_y_name() <<
+                                  "\": " << mfxVPPParams.vpp.In.CropY <<
+                                  " must be less or equal to \"" <<
+                                  CfgParam::vpp_in_height_name() << "\": " <<
+                                  mfxVPPParams.vpp.In.Height);
+        GAPI_Assert(false && "Invalid VPP params combination: Height & Crop");
+    }
+
+    if (mfxVPPParams.vpp.Out.Width < mfxVPPParams.vpp.Out.CropW + mfxVPPParams.vpp.Out.CropX) {
+        GAPI_LOG_WARNING(nullptr, "Invalid vonfiguration params: sum \"" <<
+                                  CfgParam::vpp_out_crop_w_name() <<
+                                  "\": " << mfxVPPParams.vpp.Out.CropW << " and \"" <<
+                                  CfgParam::vpp_out_crop_x_name() <<
+                                  "\": " << mfxVPPParams.vpp.Out.CropX <<
+                                  " must be less or equal to \"" <<
+                                  CfgParam::vpp_out_width_name() << "\": " <<
+                                  mfxVPPParams.vpp.Out.Width);
+        GAPI_Assert(false && "Invalid VPP params combination: Width & Crop");
+    }
+
+    if (mfxVPPParams.vpp.Out.Height < mfxVPPParams.vpp.Out.CropH + mfxVPPParams.vpp.Out.CropY) {
+        GAPI_LOG_WARNING(nullptr, "Invalid vonfiguration params: sum \"" <<
+                                  CfgParam::vpp_out_crop_h_name() <<
+                                  "\": " << mfxVPPParams.vpp.Out.CropH << " and \"" <<
+                                  CfgParam::vpp_out_crop_y_name() <<
+                                  "\": " << mfxVPPParams.vpp.Out.CropY <<
+                                  " must be less or equal to \"" <<
+                                  CfgParam::vpp_out_height_name() << "\": " <<
+                                  mfxVPPParams.vpp.Out.Height);
+        GAPI_Assert(false && "Invalid VPP params combination: Height & Crop");
+    }
+
+    GAPI_LOG_INFO(nullptr, "Finished VPP param validation");
+}
+
+ProcessingEngineBase::ExecutionStatus VPLLegacyTranscodeEngine::execute_op(operation_t& op, EngineSession& sess) {
+    return op(sess);
+}
```

Comment:
Does this method not derived from `VPLLegacyDecodeEngine`? It does there the same thing.  
It's protected there, but is it important to make it private in `VPLLegacyTranscodeEngine`?

EDIT: Why is the method even `virtual` in `ProcessingEngineBase` in the first place? It seems doing the same thing in all `ProcessingEngineBase`,  `VPLLegacyDecodeEngine` and `VPLLegacyTranscodeEngine`

Reply:
i don't remember sorry...
may be in unit tests it is used as public.
I'll try to examine it